tests/benchmark_helpers.rs

```diff
@@ -0,0 +1,115 @@
+//! Simple tests for benchmark helper functions
+//! These ensure the benchmark code itself is correct
+
+/// Simple O(n*m) pattern matcher - copy from benchmarks
+fn naive_search(haystack: &[u8], needle: &[u8]) -> Option<usize> {
```

Comment:
This is not the function used in the actual program, just a copy. Please try to use real functions.
Move related tests into the src/scanner file instead.